web/src/builder.rs

```diff
@@ -340,53 +341,104 @@ impl RuffleInstanceBuilder {
 impl RuffleInstanceBuilder {
     pub fn setup_fonts(&self, player: &mut Player) {
         for (font_name, bytes) in &self.custom_fonts {
-            if let Ok(swf_stream) = swf::decompress_swf(&bytes[..]) {
-                if let Ok(swf) = swf::parse_swf(&swf_stream) {
-                    let encoding = swf::SwfStr::encoding_for_version(swf.header.version());
-                    for tag in swf.tags {
-                        match tag {
-                            swf::Tag::DefineFont(_font) => {
-                                tracing::warn!("DefineFont1 tag is not yet supported by Ruffle, inside font swf {font_name}");
-                            }
-                            swf::Tag::DefineFont2(font) => {
-                                tracing::debug!(
-                                    "Loaded font {} from font swf {font_name}",
-                                    font.name.to_str_lossy(encoding)
-                                );
-                                player
-                                    .register_device_font(FontDefinition::SwfTag(*font, encoding));
-                            }
-                            swf::Tag::DefineFont4(font) => {
-                                let name = font.name.to_str_lossy(encoding);
-                                if let Some(data) = font.data {
-                                    tracing::debug!("Loaded font {name} from font swf {font_name}");
-                                    player.register_device_font(FontDefinition::FontFile {
-                                        name: name.to_string(),
-                                        is_bold: font.is_bold,
-                                        is_italic: font.is_italic,
-                                        data: data.to_vec(),
-                                        index: 0,
-                                    })
-                                } else {
-                                    tracing::warn!(
-                                        "Font {name} from font swf {font_name} contains no data"
+            let bytes_slice = &bytes[..];
+            if let Ok(face) = ttf_parser::Face::parse(bytes_slice, 0) {
+                tracing::debug!("Loading font {font_name} as TTF/OTF/TTC/OTC font");
+
+                // Check if font collection
+                let number_of_fonts = ttf_parser::fonts_in_collection(bytes_slice).unwrap_or(1u32);
+
+                Self::register_ttf_face_by_name(font_name, bytes.clone(), face, 0, player);
+
+                // Register all remaining fonts in the collection if it is a collection
+                for i in 1u32..number_of_fonts {
+                    let face = ttf_parser::Face::parse(bytes_slice, i)
```

Comment:
We should not panic for unparsable faces.

Reply:
Done